src/context/WalletProvider/NativeWallet/components/LegacyTwoFactor.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  ModalBody,
+  ModalHeader,
+} from '@chakra-ui/react'
+import { FieldValues, useForm } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router-dom'
+import { Text } from 'components/Text'
+
+import { LocationState } from '../types'
+
+export const LegacyTwoFactor = () => {
+  const history = useHistory()
+  const location = useLocation<LocationState>()
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors, isSubmitting },
+  } = useForm({ shouldUnregister: true })
+
+  const translate = useTranslate()
+
+  const onSubmit = async (values: FieldValues) => {
+    history.push('/native/legacy/login/success', { vault: location.state.vault })
```

Comment:
The vault won't exist yet at this point, we'll be doing another api call once they enter the 2fa. I'd structure this the same as the login onSubmit.

```suggestion
    try {
      await new Promise((resolve, reject) => resolve(null))
      // TODO: use response to create wallet vault.
      const vault = await Vault.create(undefined, false)
      vault.meta.set('createdAt', Date.now())
      vault.set('#mnemonic', DUMMY_MNEMONIC)
      history.push('/native/legacy/login/success', { vault })
    } catch (err) {
      if (isLoginError(err) && err.message === '2fa required') {
        history.push('/native/legacy/two-factor')
      }
      // TODO: show login error
    }```